pkg/autoscaler/autoscaler.go

```diff
@@ -468,18 +541,27 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 			cleanupObjects[workload.GetName()] = workload
 			Expect(client.Create(ctx, workload)).Should(Succeed(), "Failed to create scale-out workload %s", uniqueJobName)
 
-			expectedReplicas := int32(2)
-			for _, machineSet := range transientMachineSets {
-				By(fmt.Sprintf("Waiting for MachineSet %s replicas to scale out", machineSet.GetName()))
-				Eventually(func() (int, error) {
-					current, err := framework.GetMachineSet(client, machineSet.GetName())
+			expectedReplicas := int(2)
+			By("Waiting for transient MachineSets replicas to scale out")
+			Eventually(func() ([]int, error) {
+				allreplicas := []int{}
+
+				for _, machineSet := range transientMachineSets {
+					ms, err := framework.GetMachineSet(client, machineSet.GetName())
 					if err != nil {
-						return 0, err
+						return allreplicas, err
 					}
 
-					return int(pointer.Int32Deref(current.Spec.Replicas, 0)), nil
-				}, framework.WaitOverMedium, pollingInterval).Should(BeEquivalentTo(expectedReplicas), "MachineSet %s failed to scale out to %d replicas", machineSet.GetName(), expectedReplicas)
-			}
+					replicas := int(pointer.Int32Deref(ms.Spec.Replicas, 0))
+					if replicas > expectedReplicas {
+						return allreplicas, StopTrying(fmt.Sprintf("exceeded expected replicas in MachineSet %s", ms.GetName()))
+					}
+
+					allreplicas = append(allreplicas, replicas)
+				}
+
+				return allreplicas, nil
+			}, framework.WaitOverMedium, pollingInterval).Should(ConsistOf([]int{expectedReplicas, expectedReplicas}), "Failed to balance properly")
```

Comment:
The following would also work
```suggestion
			}, framework.WaitOverMedium, pollingInterval).Should(HaveEach(expectedReplicas), "Failed to balance properly")
```

Reply:
i wanted to use `HaveExactElements` , but perhaps we need to bump our version of gomega because my compiler wasn't finding it.

https://onsi.github.io/gomega/#haveexactelementselement-interface